_docs/rfc/scaffolding_new_modules.md

```diff
@@ -0,0 +1,402 @@
+through and update all the variable references to reflect what we want in the QA environment, such as the name of
+resources and instance sizes.
+
+Adding a new region or account is similar. For a new region, instead of copying `stage/us-east-1/stage` you copy
+`stage/us-east-1`. For a new account, you copy `stage`.
+
+This procedure is similar to the promote step outlined in [Adding a new component](#adding-a-new-component). I won't go
+through the details here since the procedure is relatively straightforward, but there are still some friction points:
+
+- Like in [Adding a new component](#adding-a-new-component), it won't be obvious what each of the variables actually
+  mean since the description is most likely not included. You will either rely on gut interpretation, your memory, or
+  deduction by name to decide which ones need to be updated for a new environment.
+- It isn't immediately obvious which variables need to actually change. In fact, you might have to update one of the
+  defaults for your new environment! In this scenario, you will have to have a good understanding of the stack and the
+  underlying `variables.tf` for each module to know this, or have the discipline to check each `variables.tf`. Both are
+  unreasonable expectations.
+
+---
+
+Each of these friction points are currently resolved by operator discipline. As an operator of Terragrunt, you need to
+make sure that each of your configs are sufficiently documented so that you know which variables need to be changed when
+you want to replicate them. Alternatively, you need to know where to look for that information and have the discipline
+to check each time. Both of these are unreasonable to expect for sufficiently large infrastructure deployments with many
+possible modules and dependencies.
+
+The goal of this RFC is to propose an approach to handling these basic tasks that minimizes the room for operator
+errors.
+
+
+## Proposed solution
+
+We will introduce a new command `terragrunt new`, which provides a code generation interface for solving each of the
+scenarios described above. The goal of `terragrunt new` is to provide the ability to templatize your configurations with
+annotations in the code that reminds the user what pieces need to change for each scenario.
+
+It is important that the templates are customizable. Not all users have the same requirements, so allowing users to
+customize the templates to their needs is important. The goal is to allow users to define their own standards as
+templates that then is used by their engineers to generate new configurations that conform to their standards, which may
+be different from Gruntwork standards. I envision Gruntwork providing a set of base templates that can be used by
+default, which can be overridden by either forking our `infrastructure-templates` repository or by building new ones
+from scratch.
+
+`terragrunt new` should support the following scenarios:
```

Comment:
We may also want:

1. `terragrunt new basic-scaffold`, which creates an infra-live and infra-modules with a reasonable basic starting structure: e.g., a basic module in infra-modules and deploying that module in a couple envs in infra-live.
1. `terragrunt new module`: creates dummy `main.tf`, `variables.tf`, `outputs.tf`, `dependencies.tf`, `README.md`, and perhaps Terratest tests for this module.